### PR TITLE
feat: enable watermark field as ENV variable in vertex

### PR DESCRIPTION
**Pull request by vigith**

### CODE REVIEW

File: pkg/apis/numaflow/v1alpha1/pipeline_types.go [hunk 1]
```diff
@@ -270,8 +270,19 @@ type PipelineSpec struct {
 	Lifecycle Lifecycle `json:"lifecycle,omitempty" protobuf:"bytes,4,opt,name=lifecycle"`
 	// Limits define the limitations such as buffer read batch size for all the vertices of a pipleine, they could be overridden by each vertex's settings
 	// +kubebuilder:default={"readBatchSize": 100, "udfWorkers": 100, "bufferMaxLength": 10000, "bufferUsageLimit": 80}
-	// +optinal
+	// +optional
 	Limits *PipelineLimits `json:"limits,omitempty" protobuf:"bytes,5,opt,name=limits"`
+	// Watermark enables watermark progression across the entire pipeline.
```
Comment (whynowy) [inline pkg/apis/numaflow/v1alpha1/pipeline_types.go:275]: Add some comments that it should be not updated after pipeline is created.

File: pkg/isb/jetstream/writer_test.go [hunk 2]
```diff
@@ -140,6 +140,8 @@ func TestForwarderJetStreamBuffer(t *testing.T) {
 
 // TestJetStreamBufferWrite on buffer full
 func TestJetStreamBufferWriterBufferFull(t *testing.T) {
+	// FIXME: #11
+	t.SkipNow()
```
Comment (whynowy) [inline pkg/isb/jetstream/writer_test.go:144]: It should be working with latest `main`.

Comment (vigith) [reply pkg/isb/jetstream/writer_test.go]: done

File: pkg/apis/numaflow/v1alpha1/pipeline_types.go (hunk 1 above)
Comment (vigith) [reply pkg/apis/numaflow/v1alpha1/pipeline_types.go]: done

**Review by whynowy — APPROVED** [review]
LGTM

**Pull request merged**